torchao/prototype/low_bit_optim/subclass_fp8.py

```diff
@@ -145,6 +149,44 @@ def _(func, types, args, kwargs):
     )
 
 
+# required by torch.distributed.checkpoint.save
+# note that we don't actually implement pin memory for this tensor subclass
+# (pin_memory argument is ignored in aten._to_copy)
+@OptimStateFp8.implements(aten.is_pinned.default)
+def _(func, types, args, kwargs):
+    return args[0].codes.is_pinned() and args[0].scale.is_pinned()
+
+
+# required by torch.distributed.checkpoint.load when world size changes i.e. re-sharding
+@OptimStateFp8.implements(aten.slice.Tensor)
+def _(func, types, args, kwargs):
+    x, dim, start, end = args[:4]
+    step = args[4] if len(args) > 4 else 1
+
+    # input validation
+    if dim != 0:
+        raise ValueError("Only support aten.slice along the first dim")
```

Comment:
@gau-nernst  this raises error when swithching from TP=1,DP=8 to DP=1,TP=8

Reply:
Because of block-wise quantization, slicing in any dim > 0 is messy. It's doable, but just messy, and it does not always work (i.e. if you are slicing in the middle of a quantization block -> not possible).
I don't use TP before so I don't know how it does the sharding. Can you try print out the `x.shape, dim, start, end`? And possibly open a new issue so we can discuss there.